src/Microsoft.AspNetCore.Rewrite/Internal/ApacheModRewrite/CookieActionFactory.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Rewrite.Internal.UrlActions;
+
+namespace Microsoft.AspNetCore.Rewrite.Internal.ApacheModRewrite
+{
+    public class CookieActionFactory
+    {
+        /// <summary>
+        ///  Creates a <see cref="ChangeCookieAction" /> <see href="https://httpd.apache.org/docs/current/rewrite/flags.html#flag_co" /> for details.
+        /// </summary>
+        /// <param name="flagValue">The flag</param>
+        /// <returns>The action</returns>
+        public ChangeCookieAction Create(string flagValue)
+        {
+            if (string.IsNullOrEmpty(flagValue))
+            {
+                throw new ArgumentNullException(nameof(flagValue));
+            }
+
+            var i = 0;
+            var separator = ':';
+            if (flagValue[0] == ';')
+            {
+                separator = ';';
+                i++;
+            }
+
+            ChangeCookieAction action = null;
+
+            var current = CookieFlagToken.Name;
+            var start = i;
+            for (; i < flagValue.Length; i++)
+            {
+                if (flagValue[i] == separator)
+                {
+                    var length = i - start;
+                    SetActionOption(flagValue.Substring(start, length).Trim(), current, ref action);
+
+                    current++;
+                    start = i + 1;
+                }
+            }
+
+            if (i != start)
+            {
+                SetActionOption(flagValue.Substring(start).Trim(new[] { ' ', separator }), current, ref action);
+            }
+
+
+            if (current < CookieFlagToken.Domain)
+            {
+                throw new FormatException(Resources.FormatError_InvalidChangeCookieFlag(flagValue));
+            }
+
+            return action;
+        }
+
+        private static void SetActionOption(string value, CookieFlagToken tokenType, ref ChangeCookieAction action)
+        {
+            switch (tokenType)
+            {
+                case CookieFlagToken.Name:
+                    action = new ChangeCookieAction(value);
+                    break;
+                case CookieFlagToken.Value:
+                    action.Value = value;
+                    break;
+                case CookieFlagToken.Domain:
+                    action.Domain = string.IsNullOrEmpty(value)
+                        ? null
+                        : value;
+                    break;
+                case CookieFlagToken.Lifetime:
+                    if (string.IsNullOrEmpty(value))
+                    {
+                        break;
+                    }
+
+                    action.Lifetime = int.Parse(value);
```

Comment:
Nah, we want this to throw. Apache would also throw if there is a bad number format.